clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteClusterResourceDefinition.java

```diff
@@ -0,0 +1,151 @@
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.CapabilityReferenceRecorder;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.StringListAttributeDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.registry.AttributeAccess;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.wildfly.clustering.service.BinaryRequirement;
+
+/**
+ * /subsystem=infinispan/remote-cache-container=X/remote-cluster=Y
+ *
+ * @author Radoslav Husar
+ */
+public class RemoteClusterResourceDefinition extends ChildResourceDefinition<ManagementResourceRegistration> {
+
+    public static final PathElement WILDCARD_PATH = pathElement(PathElement.WILDCARD_VALUE);
+
+    public static PathElement pathElement(String name) {
+        return PathElement.pathElement("remote-cluster", name);
+    }
+
+    public enum Attribute implements org.jboss.as.clustering.controller.Attribute {
+        SOCKET_BINDINGS("socket-bindings", new CapabilityReference(Capability.REMOTE_CLUSTER, CommonUnaryRequirement.OUTBOUND_SOCKET_BINDING)),
+        ;
+
+        private final AttributeDefinition definition;
+
+        Attribute(String name, CapabilityReferenceRecorder reference) {
+            this.definition = new StringListAttributeDefinition.Builder(name)
+                    .setAllowExpression(false)
+                    .setRequired(true)
+                    .setCapabilityReference(reference)
+                    .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
+                    .build();
+        }
+
+        @Override
+        public AttributeDefinition getDefinition() {
+            return this.definition;
+        }
+    }
+
+     enum Requirement implements BinaryRequirement, BinaryServiceNameFactoryProvider {
+        REMOTE_CLUSTER("org.wildfly.clustering.infinispan.remote-cache-container.remote-cluster", Void.class),
+        ;
+        private final String name;
+        private final Class<?> type;
+        private final BinaryServiceNameFactory factory = new BinaryRequirementServiceNameFactory(this);
+
+         Requirement(String name, Class<?> type) {
+            this.name = name;
+            this.type = type;
+        }
+
+        @Override
+        public String getName() {
+            return this.name;
+        }
+
+        @Override
+        public Class<?> getType() {
+            return this.type;
+        }
+
+        @Override
+        public BinaryServiceNameFactory getServiceNameFactory() {
+            return this.factory;
+        }
+    }
+
+
+    enum Capability implements org.jboss.as.clustering.controller.Capability {
+        REMOTE_CLUSTER("org.wildfly.clustering.infinispan.remote-cache-container.remote-cluster", Void.class),
```

Comment:
Make sure to add this to wildfly-capabilities documentation.

Reply:
PR open https://github.com/wildfly/wildfly-capabilities/pull/20